workflows/benchmarks/benchmark-analysis.Rmd

```diff
@@ -0,0 +1,324 @@
+
+## Comparing decoy indexes
+
+Get cDNA-only decoy comparison data.
+```{r}
+sce_feature_filtered <- sce_feature_qc %>% 
+  dplyr::filter(kmer == "31", index_content == "cdna")
+```
+
+```{r}
+decoy_means <- sce_feature_filtered %>%
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = decoy,
+                     values_from = mean) 
+```
+
+First off, how well correlated are the means for various decoy choices? 
+We expect very good correlations.
+
+```{r}
+cor(decoy_means[,3:5], method = "spearman", use = "complete.obs")
+```
+
+
+
+And we have them... interestingly, the best correlation is between no decoys and the partial decoy: the full decoy index is more different by this measure.
+This leads me to the preliminary conclusion that the partial index (which notably uses a slightly different set of transcripts at this stage, due to not being built locally) may not be worth pursuing, as it seems to make very little difference.
+
+Don't forget to look at your data!
+```{r}
+pairs(decoy_means[,3:5])
+```
+Whoa... seems like a lot is being driven by a few points that are very different in the full index data. 
+What are those genes?
+
+
+```{r}
+decoy_means %>% 
+  dplyr::mutate(diff = (no - full)/(full + 1)) %>%
+  dplyr::arrange(desc(diff))
+```
+ENSG00000269028, which is by far the most different, is the gene *MT-RNR2 like 12* which, aside from being related to at least 12 other genes, does not ring any particular bells.
+However, ENSG00000255823 is *MT-RNR2-like 8* 
+
+Other genes at the top of the differential list:
+ENSG00000163864: NMNAT3 nicotinamide nucleotide adenylyltransferase 3
+ENSG00000197563: PIGN Phosphatidylinositol Glycan Anchor Biosynthesis Class N
+ENSG00000173559: NABP1 Nucleic Acid Binding Protein 1
+
+I don't really know what to do with this list... it seems like few genes will have large differences, but a few differences are enormous!
+
+
+Just as a sanity check, lets repeat one of the analyses we did before, comparing the cDNA to the transcriptome for the full decoy index (since we now have both):
+
+```{r}
+# using the same `cell_sample` and `coding_features` as previously
+
+compare_expression_full <- data.frame(
+  cdna = as.vector(counts(sces[["905_3-cdna_k31_full_sa"]][coding_features, cell_sample])),
+  txome = as.vector(counts(sces[["905_3-txome_k31_full_sa"]][coding_features, cell_sample]))
+)
+
+cor(compare_expression_full, method = "spearman")
+```
+```{r}
+ggplot(compare_expression_full, aes(log1p(cdna), y = log1p(txome))) +
+  geom_point()
+```
+
+Interestingly, the correlation between cDNA and txome expression values (for coding genes) using the full decoy seems substantially better than with the no decoy sequences!
+
+This seems to me to be a good argument for using the full decoy, if it is less sensitive to the chosen transcript list.
```

Comment:
Can we stick a `sessionInfo()` here? I know this should always be run in a container, but this practice has been helpful for tracking down potential weirdness in the past.